contracts/YVaultV4AssetProxy.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+// WARNING: This has been validated for yearn vaults version 4.2, do not use for lower or higher
+//          versions without review
+pragma solidity ^0.8.0;
+
+import "./interfaces/IERC20.sol";
+import "./interfaces/IYearnVault.sol";
+import "./YVaultAssetProxy.sol";
+
+/// @author Element Finance
+/// @title Yearn Vault Asset Proxy
+contract YVaultV4AssetProxy is YVaultAssetProxy {
+    /// @notice Constructs this contract by calling into the super constructor
+    /// @param vault_ The yearn v2 vault, must be version 0.4.2
+    /// @param _token The underlying token.
+    ///               This token should revert in the event of a transfer failure.
+    /// @param _name The name of the token created
+    /// @param _symbol The symbol of the token created
+    constructor(
+        address vault_,
+        IERC20 _token,
+        string memory _name,
+        string memory _symbol
+    ) YVaultAssetProxy(vault_, _token, _name, _symbol) {}
+
+    /// @notice Overrides the version checking to check for 0.4.2 instead
+    /// @param _vault The yearn vault address
+    /// @dev This function can be overridden by an inheriting upgrade contract
+    function _versionCheck(IYearnVault _vault) internal override view {
+        string memory apiVersion = _vault.apiVersion();
+        require(_stringEq(apiVersion, "0.4.2"), "Unsupported Version");
```

Comment:
If they release a v4.3+ vault that is compatible with this proxy logic, then we still have to change this versionCheck and redeploy, right?

Reply:
We won't have to redeploy anything since the user proxies are not deployed via factory, but we will need to put in an upgrade before we deploy on that version